### DISCUSSION TITLE
ThingsBoard deploy with source code

### DISCUSSION THREAD

**Discussion by muhammetcetinyamac** [thingsboard/thingsboard#10866] [Q&A]
Hello
I want to build and run Thingsboard with its source code in a Docker environment. Has anyone written a sample base Dockerfile for this before?
Thank you.

**Comment by ViacheslavKlimov** [comment]
Hi @muhammetcetinyamac,

If you run `mvn clean license:format install -DskipTests -Ddockerfile.skip=false`, it will build all the needed docker images locally from the source code.
